src/scenic/core/propositions.py

```diff
@@ -0,0 +1,199 @@
+"""Objects representing propositions that can be used to specify conditions"""
+
+from functools import reduce
+import operator
+import rv_ltl
+from scenic.core.errors import InvalidScenarioError
+
+from scenic.core.lazy_eval import needsLazyEvaluation
```

Comment:
A very minor stylistic point: I like to group top-level imports into 3 sections, separated by blank lines: Python standard library imports, other external imports, then imports from Scenic itself. This helps make clear which external/internal dependencies the file has.

Reply:
Fixed in fc851b08d425ac04d7bcd85f255be102e59d691a